walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +359,54 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void optionalAttendeesNoGaps () {
+    // Have two optional attendees with no gaps in schedule. Return full day since mandatory attendees are free all day. 
+    //
+    // Optional: |--A-----||-----B-----|
+    // Day     : |---------------------|
+    // Options : |---------------------|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0830AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_60_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(TimeRange.WHOLE_DAY);
+
+    Assert.assertEquals(expected, actual);
+
+  }
+
+  @Test
+  public void optionalAttendeesWithGaps() {
+    // Have two optional attendees with gaps in schedule. Return those gaps. 
+    //
+    // Optional: |--A--|        |--B---|
+    // Day     : |---------------------|
+    // Options :
```

Comment:
Update this part of the diagram

Reply:
I was trying to show that there are no options! I just made it exclamation points instead and made a clearer comment

Reply:
Isn't (8:30 - 11:00) the available option, from line 407? Maybe this commit is outdated and we're looking at different things.

Reply:
OH I was looking at the wrong function, sorry, I've fixed it now